File: 251001/Churavskiy/lab3/discussion/src/main/java/com/example/discussion/repository/ReactionRepository.java
package com.example.discussion.repository;

import com.example.discussion.entity.Reaction;
import com.example.discussion.entity.ReactionKey;
import org.springframework.data.cassandra.repository.CassandraRepository;
import org.springframework.data.cassandra.repository.Query;
import org.springframework.stereotype.Repository;

import java.util.List;
import java.util.Optional;

@Repository
public interface ReactionRepository extends CassandraRepository<Reaction, ReactionKey> {
    @Query("SELECT * FROM tbl_reaction WHERE id = :id ALLOW FILTERING")
    Optional<Reaction> findFirstById(Long id);

    @Query("DELETE FROM tbl_reaction WHERE id = :id ALLOW FILTERING")
    void deleteById(Long id);
}

File: 251001/Vasilenko/discussion/src/main/kotlin/org/ex/discussion/repository/ReactionRepository.kt
package org.ex.discussion.repository

import org.ex.discussion.model.Reaction
import org.springframework.data.cassandra.repository.Query
import org.springframework.data.repository.CrudRepository
import org.springframework.data.repository.query.Param
import org.springframework.stereotype.Repository
import java.util.*

@Repository
interface ReactionRepository : CrudRepository<Reaction, Long> {

    @Query(
        value = "SELECT * FROM reaction WHERE id = :id ALLOW FILTERING;",
        allowFiltering = true
    )
    override fun findById(@Param("id") id: Long): Optional<Reaction>

    @Query(
        value = "DELETE FROM reaction WHERE id = :id AND country = :country;"
    )
    fun delete(@Param("id") id: Long, @Param("country") country: String)
}